src/pages/home/report/ReportActionItemMessageEdit.js

```diff
@@ -156,14 +154,14 @@ class ReportActionItemMessageEdit extends React.Component {
      * @param {String} emoji
      */
     addEmojiToTextBox(emoji) {
-        const newComment = this.state.draft.slice(0, this.state.selection.start)
-            + emoji + this.state.draft.slice(this.state.selection.end, this.state.draft.length);
-        this.setState(prevState => ({
-            selection: {
-                start: prevState.selection.start + emoji.length,
-                end: prevState.selection.start + emoji.length,
-            },
-        }));
+        const newComment = this.draft.slice(0, this.selection.start)
+            + emoji + this.draft.slice(this.selection.end, this.draft.length);
+        this.selection = {
+            start: this.selection.start + emoji.length,
+            end: this.selection.start + emoji.length,
+        };
+
+        this.textInput.setTextAndSelection(newComment, this.selection.start, this.selection.end);
```

Comment:
Related to my previous comment - it might be better to have a lib `viewCommands` or `textCommands`, import it and use it here, instead of adding a `setTextAndSelection` method to the `textInput` ref